lua/vgit/features/buffer/LiveGutter.lua

```diff
@@ -44,6 +44,8 @@ LiveGutter.sync = loop.debounce(
     if not hunks then
       buffer:set_cached_live_signs(live_signs)
       return
+    else
+      buffer:store_buffer_vars()
```

Comment:
Using the solution from `Buffer.lua` I would then add the following here:

```lua
local diff_status = buffer.git_object:generate_diff_status()

loop.await_fast_event() -- vim.schedule
buffer:set_var('vgit_status', diff_status)
```

The plenary lib is awesome and it helps me get rid of all callbacks within the plugin :) and so `loop.await_fast_event()` should help us keep things linear.

Reply:
done